packages/itwinui-css/src/table/base.scss

```diff
@@ -22,10 +32,11 @@
 }
 
 @mixin iui-table-header-wrapper {
-  @include table-scrollbar;
   scrollbar-color: transparent transparent; // Reserve space but never show the scrollbar
-  overflow-x: hidden;
-  background-color: var(--iui-color-background);
+  position: sticky;
+  inset-block-start: 0;
+  z-index: 2;
```

Comment:
Comments would be useful next to z-index

Reply:
Added comments and removed an unnecessary z-index (https://github.com/iTwin/iTwinUI/pull/1891/commits/6e68ef36e81698c8bb3ebe5c4f63270457db3da7).

Reply:
Fixed the z-index that regressed after https://github.com/iTwin/iTwinUI/commit/6e68ef36e81698c8bb3ebe5c4f63270457db3da7 and updated the comment accordingly (https://github.com/iTwin/iTwinUI/pull/1891/commits/1d32c730da4b7773d4899a59efb098823ef3df7f).

Reply:
Fixed the z-index that regressed after https://github.com/iTwin/iTwinUI/commit/6e68ef36e81698c8bb3ebe5c4f63270457db3da7 and updated the comment accordingly (https://github.com/iTwin/iTwinUI/pull/1891/files/730397d6ee1fac7d0275263ad1f15c9e3c6205fa..a52efb2654dbf217cb95af7e60e6a13bcd50f239).

Reply:
Fixed another regression (https://github.com/iTwin/iTwinUI/pull/1891/commits/d14a94c2557b6546f39558cdedf31ad14516fc55).

Reply:
Visual tests coming in clutch